File: java/src/strategy/SimUDuckApp/Revision1/Duck.java
package strategy.SimUDuckApp.Revision1;

public abstract class Duck {

	public void quack() {
		System.out.println("Quack");
	}
	
	public void swim() {
		System.out.println("All ducks float, even decoys!");
	}
	abstract public void display();
	public void fly() {
		System.out.println("I'm flying");
	}
}

File: src/strategy/SimUDuckApp/Revision1/Duck.kt
package strategy.SimUDuckApp.Revision1

abstract class Duck {
    fun quack() {
        println("Quack")
    }

    fun swim() {
        println("All ducks float, even decoys!")
    }

    abstract fun display()
    fun fly() {
        println("I'm flying")
    }
}
